moveos/moveos-stdlib/moveos-stdlib/sources/object.move

```diff
@@ -2,159 +2,88 @@
 module moveos_std::object {
-    use std::bcs;
-    use moveos_std::address;
     use moveos_std::tx_context::{Self, TxContext};
-
-    /// An object ID. This is used to reference Sui Objects.
-    /// This is *not* guaranteed to be globally unique--anyone can create an `ID` from a `UID` or
-    /// from an object, and ID's can be freely copied and dropped.
-    /// Here, the values are not globally unique because there can be multiple values of type `ID`
-    /// with the same underlying bytes. For example, `object::id(&obj)` can be called as many times
-    /// as you want for a given `obj`, and each `ID` value will be identical.
-    struct ID has copy, drop, store {
-        // We use `address` instead of `vector<u8>` here because `address` has a more
-        // compact serialization. `address` is serialized as a BCS fixed-length sequence,
-        // which saves us the length prefix we would pay for if this were `vector<u8>`.
-        // See https://github.com/diem/bcs#fixed-and-variable-length-sequences.
-        bytes: address
-    }
-
-    /// Globally unique IDs that define an object's ID in storage. Any Sui Object, that is a struct
-    /// with the `key` ability, must have `id: UID` as its first field.
-    /// These are globally unique in the sense that no two values of type `UID` are ever equal, in
-    /// other words for any two values `id1: UID` and `id2: UID`, `id1` != `id2`.
-    /// This is a privileged type that can only be derived from a `TxContext`.
-    /// `UID` doesn't have the `drop` ability, so deleting a `UID` requires a call to `delete`.
-    struct UID has store {
-        id: ID,
-    }
-
-    // === id ===
-
-    /// Get the raw bytes of a `ID`
-    public fun id_to_bytes(id: &ID): vector<u8> {
-        bcs::to_bytes(&id.bytes)
+    use moveos_std::any_table::{Self};
+    
+    /// Invalid access of object, the object is not owned by the signer or the object is not shared or immutable
+    const EInvalidAccess: u64 = 0;
+   
+    struct ObjectID has store, copy, drop {
+        //TODO should use u256 to replace address?
+        id: address,
     }
 
-    /// Get the inner bytes of `id` as an address.
-    public fun id_to_address(id: &ID): address {
-        id.bytes
+    /// Box style object
+    /// The object can not be copied, droped, only can be consumed by `add`
+    struct Object<T> {
+        id: ObjectID,
+        value: T,
+        owner: address,
+        //TODO define shared and immutable
+        //shared: bool,
+        //immutable: bool,
     }
 
-    /// Make an `ID` from raw bytes.
-    public fun id_from_bytes(bytes: vector<u8>): ID {
-        id_from_address(address::from_bytes(bytes))
+    #[private_generic(T)]
+    /// Create a new object, the object is owned by `owner`
+    /// The private generic is indicate the T should be defined in the same module as the caller. This is ensured by the verifier.
+    public fun new<T: key>(ctx: &mut TxContext, owner: address, value: T): Object<T>{
```

Comment:
When the address corresponding to the signer in the owner and tx_context is inconsistent, it is equivalent to implicitly operating the transfer after creating the object, and the owner passes the parameter to become the owner of the object?

Reply:
The object owner can be a resource account, so let the caller decide who is owner.